src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
Note that we are currently very small set of data. if we add more and more such tests we'll run into such issues all the time, Also I am expecting in the near future we'll run the tests on different versions of the same OS and at that time we'll have to make the test more complicated to handle all possible versions we run on. we are currently suffering from this way of testing in the desktop because of the legacy tests written assuming some specific data and that is not valid anymore. the cost maintaining such test is really high especially when the test cover more cases. 
if we make the test read the OS data we'll not need to visit the issue again. if we have any bug in reading the data, that most likely will be reported and we'll fix it once for good.